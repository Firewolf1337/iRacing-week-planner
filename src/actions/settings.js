import debounce from 'lodash.debounce';

export const UPDATE_FILTERS = 'SETTINGS/UPDATE_FILTERS';
export const RESET_FILTERS = 'SETTINGS/RESET_FILTERS';
export const RESET_SETTINGS = 'SETTINGS/RESET_SETTINGS';
export const UPDATE_SETTING = 'SETTINGS/UPDATE_SETTING';
export const LOAD_SETTINGS_FROM_FIREBASE = 'SETTINGS/LOAD_SETTINGS_FROM_FIREBASE';
export const FIREBASE_SYNCED = 'SETTINGS/FIREBASE_SYNCED';

export function saveSettingsToFirebase() {
  return async (dispatch, getState) => {
    const { user, firebaseApp } = getState().auth;
    const { settings } = getState();

    if (!user) {
      return;
    }

    const db = firebaseApp.firestore();
    const docRef = db.collection('settings').doc(user.uid);
    await docRef.set(settings);
    dispatch({ type: FIREBASE_SYNCED });
  };
}

const dispatchSaveSettingsToFirebase = (dispatch) => {
  dispatch(saveSettingsToFirebase());
};

export const debouncedDispatcherSaveSettings = debounce(dispatchSaveSettingsToFirebase, 10000);

export function updateFilters(newFilters) {
  return async (dispatch) => {
    await dispatch({ type: UPDATE_FILTERS, payload: { filters: newFilters } });
    debouncedDispatcherSaveSettings(dispatch);
  };
}

export function resetFilters() {
  return async (dispatch) => {
    await dispatch({ type: RESET_FILTERS, payload: {} });
    debouncedDispatcherSaveSettings(dispatch);
  };
}

export function resetSettings() {
  return async (dispatch) => {
    await dispatch({ type: RESET_SETTINGS });
    debouncedDispatcherSaveSettings(dispatch);
  };
}

export function updateSetting(key, value) {
  return async (dispatch) => {
    await dispatch({ type: UPDATE_SETTING, payload: { key, value } });
    debouncedDispatcherSaveSettings(dispatch);
  };
}

export function getSettingsFromFirebase() {
  return async (dispatch, getState) => {
    const { user, firebaseApp } = getState().auth;
    if (!user) {
      return;
    }

    const db = firebaseApp.firestore();
    const docRef = db.collection('settings').doc(user.uid);
    const doc = await docRef.get();

    if (!doc.exists) {
      dispatch(saveSettingsToFirebase());
      return;
    }

    dispatch({ type: LOAD_SETTINGS_FROM_FIREBASE, payload: doc.data() });
  };
}
