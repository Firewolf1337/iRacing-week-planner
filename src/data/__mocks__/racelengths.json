{
  "2915": {
    "0": {
      "laps": "15"
    },
    "1": {
      "laps": "25"
    },
    "2": {
      "laps": "15"
    },
    "3": {
      "laps": "15"
    },
    "4": {
      "laps": "15"
    },
    "5": {
      "laps": "20"
    },
    "6": {
      "laps": "15"
    },
    "7": {
      "laps": "15"
    },
    "8": {
      "laps": "15"
    },
    "9": {
      "laps": "25"
    },
    "10": {
      "laps": "20"
    },
    "11": {
      "laps": "15"
    }
  },
  "2948": {
    "0": {
      "minutes": "15"
    },
    "1": {
      "minutes": "15"
    },
    "2": {
      "minutes": "15"
    },
    "3": {
      "minutes": "15"
    },
    "4": {
      "minutes": "15"
    },
    "5": {
      "minutes": "15"
    },
    "6": {
      "minutes": "15"
    },
    "7": {
      "minutes": "15"
    },
    "8": {
      "minutes": "15"
    },
    "9": {
      "minutes": "15"
    },
    "10": {
      "minutes": "15"
    },
    "11": {
      "minutes": "15"
    }
  },
  "2917": {
    "0": {
      "laps": "15"
    },
    "1": {
      "laps": "25"
    },
    "2": {
      "laps": "15"
    },
    "3": {
      "laps": "15"
    },
    "4": {
      "laps": "15"
    },
    "5": {
      "laps": "20"
    },
    "6": {
      "laps": "15"
    },
    "7": {
      "laps": "15"
    },
    "8": {
      "laps": "15"
    },
    "9": {
      "laps": "25"
    },
    "10": {
      "laps": "20"
    },
    "11": {
      "laps": "15"
    }
  },
  "2923": {
    "0": {
      "laps": "15"
    },
    "1": {
      "laps": "25"
    },
    "2": {
      "laps": "15"
    },
    "3": {
      "laps": "15"
    },
    "4": {
      "laps": "15"
    },
    "5": {
      "laps": "20"
    },
    "6": {
      "laps": "15"
    },
    "7": {
      "laps": "15"
    },
    "8": {
      "laps": "15"
    },
    "9": {
      "laps": "25"
    },
    "10": {
      "laps": "20"
    },
    "11": {
      "laps": "15"
    }
  }
}
