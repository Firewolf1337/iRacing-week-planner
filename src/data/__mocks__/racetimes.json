[
  {
    "seriesId": 139,
    "everytime": {
      "minutes": 60
    },
    "offset": {
      "minutes": 30
    }
  },
  {
    "seriesId": 432,
    "setTimes": [
      {
        "minutes": 3000
      },
      {
        "minutes": 4020
      },
      {
        "minutes": 5880
      },
      {
        "minutes": 8100
      }
    ]
  },
  {
    "seriesId": 32,
    "everytime": {
      "minutes": 120
    },
    "offset": {
      "minutes": 115
    }
  },
  {
    "seriesId": 167,
    "everytime": {
      "minutes": 30
    },
    "offset": {
      "minutes": 0
    }
  }
]
