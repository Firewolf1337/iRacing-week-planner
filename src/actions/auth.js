import { debouncedDispatcherSaveSettings, getSettingsFromFirebase, saveSettingsToFirebase } from './settings';

export const LOADING_AUTH = 'AUTH/LOADING_SIGN_IN';
export const ERROR_AUTH = 'AUTH/ERROR_AUTH';
export const ERROR_RESET = 'AUTH/ERROR_RESET';
export const ERROR_ACKNOWLEDGE = 'AUTH/ERROR_ACKNOWLEDGE';
export const SIGNED_IN = 'AUTH/SIGNED_IN';
export const SIGNED_OUT = 'AUTH/SIGNED_OUT';
export const RESET_SENT = 'AUTH/RESET_SENT';
export const LOADING_RESET = 'AUTH/LOADING_RESET';

export function signOut() {
  return async (dispatch, getState) => {
    dispatch({ type: LOADING_AUTH });
    if (getState().settings.firebaseSynced === false) {
      debouncedDispatcherSaveSettings.cancel();
      await dispatch(saveSettingsToFirebase());
    }

    await getState().auth.firebaseApp.auth().signOut();
    dispatch({ type: SIGNED_OUT });
  };
}

export function signIn(email, password) {
  return async (dispatch, getState) => {
    dispatch({ type: LOADING_AUTH });
    try {
      await getState().auth.firebaseApp.auth().signInWithEmailAndPassword(email, password);
      return {};
    } catch (error) {
      dispatch({ type: ERROR_AUTH, error });
      return { type: ERROR_AUTH };
    }
  };
}

export function createAccount(email, password) {
  return async (dispatch, getState) => {
    dispatch({ type: LOADING_AUTH });
    try {
      await getState().auth.firebaseApp.auth().createUserWithEmailAndPassword(email, password);
      return {};
    } catch (error) {
      dispatch({ type: ERROR_AUTH, error });
      return { type: ERROR_AUTH };
    }
  };
}

export function acknowledgeAuthError() {
  return { type: ERROR_ACKNOWLEDGE };
}

export function signedIn(user) {
  return async (dispatch) => {
    await dispatch({ type: SIGNED_IN, user });

    if (user) {
      dispatch(getSettingsFromFirebase());
    }
  };
}

export function startListener() {
  return async (dispatch, getState) => {
    const { currentUser } = getState().auth.firebaseApp.auth();
    dispatch(signedIn(currentUser));

    getState().auth.firebaseApp.auth().onAuthStateChanged((user) => {
      dispatch(signedIn(user));
    });
  };
}

export function forgottenPassword(email) {
  return async (dispatch, getState) => {
    dispatch({ type: LOADING_RESET });
    try {
      await getState().auth.firebaseApp.auth().sendPasswordResetEmail(email);
      dispatch({ type: RESET_SENT });
      return { type: RESET_SENT };
    } catch (error) {
      dispatch({ type: ERROR_RESET, error });
      return { type: ERROR_RESET };
    }
  };
}
