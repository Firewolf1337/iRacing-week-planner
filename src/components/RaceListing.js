// @flow

import * as React from 'react';
import moment from 'moment';
import { useTranslation } from 'react-i18next';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import { updateSetting } from '../actions/settings';
import SortArrow from './SortArrow';

import allRaces from '../lib/races';
import filterRaces from '../lib/filterRaces';
import sortRaces from '../lib/sortRaces';

import availableColumns from '../data/availableColumns';

import styles from '../styles/main.module.scss';
import raceListingStyles from './styles/raceListing.module.scss';
import columnStyles from './columns/styles/columns.module.scss';

const settingsSelector = (state) => state.settings;
const dateSelector = (state) => state.app.date;
const dateSelector2 = (state) => state.app.date2;
const weekSelector = (state) => state.app.week;
const weekSelector2 = (state) => state.app.week2;

export default function RaceListing(): React.Node {
  const { t } = useTranslation();
  const settings = useSelector(settingsSelector, shallowEqual);
  const date = useSelector(dateSelector, shallowEqual);
  const date2 = useSelector(dateSelector2, shallowEqual);
  const week = useSelector(weekSelector, shallowEqual);
  const week2 = useSelector(weekSelector2, shallowEqual);
  const {
    filters, favouriteSeries, favouriteCars, favouriteTracks,
    sort, ownedCars, ownedTracks, columns,
  } = settings;
  const dispatch = useDispatch();

  const getSortColumnHandler = (columnId: string) => () => {
    if (sort.key === columnId) {
      dispatch(updateSetting('sort', { ...sort, order: sort.order === 'asc' ? 'desc' : 'asc' }));
      return;
    }

    dispatch(updateSetting('sort', { key: columnId, order: 'asc' }));
  };

  const dateFilteredRaces = allRaces.filter((race) => race.week >= week-1 && race.week <= week2-1);

  const sortedRaces = React.useMemo(() => sortRaces(sort, dateFilteredRaces), [sort, dateFilteredRaces]);

  const filteredRaces = React.useMemo(() => filterRaces({
    races: sortedRaces, filters, ownedTracks, ownedCars, favouriteSeries, favouriteCars, favouriteTracks,
  }), [sortedRaces, filters, ownedTracks, ownedCars, favouriteSeries, favouriteCars, favouriteTracks]);

  if (filters.favouriteSeries && filteredRaces.length === 0) {
    return <p>{t('No races this week match your favourite series. Try turning the filter off or adding some.')}</p>;
  }

  if (filters.favouriteCarsOnly && filteredRaces.length === 0) {
    return <p>{t('No races this week match your favourite cars. Try turning the filter off or adding some.')}</p>;
  }

  if (filters.favouriteTracksOnly && filteredRaces.length === 0) {
    return <p>{t('No races this week match your favourite tracks. Try turning the filter off or adding some.')}</p>;
  }

  const chosenColumns = availableColumns.filter((column) => columns.indexOf(column.id) !== -1);

  return (
    <div className={`${styles['table-responsive']} ${raceListingStyles.raceListing}`}>
      <table className={styles.table} style={{ fontSize: '0.8em' }}>
        <thead>
          <tr>
            {chosenColumns.map((column) => {
              const modalOpen = getSortColumnHandler(column.id);

              const content = (
                <>
                  {t(column.header)}
                  <span> </span>
                  {sort.key === column.id ? <SortArrow sort={sort}/> : null}
                </>
              );

              return (
                <th
                  key={column.id}
                  id={`raceListing-th-${column.id}`}
                  className={column.sort ? columnStyles.clickableCell : null}
                >
                  {column.sort ? (
                    <button
                      type="button"
                      className={columnStyles.cellButton}
                      onClick={modalOpen}
                      onKeyPress={modalOpen}
                    >
                      {content}
                    </button>
                  ) : content}
                </th>
              );
            })}
          </tr>
        </thead>
        <tbody>
          {filteredRaces.map((race) => (
            <tr key={race.seriesuniq}>
              {chosenColumns.map((column) => (
                <column.component
                  key={column.id}
                  race={race}
                  ownedCars={ownedCars}
                  favouriteCars={favouriteCars}
                  ownedTracks={ownedTracks}
                  favouriteTracks={favouriteTracks}
                  favouriteSeries={favouriteSeries}
                />
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}
