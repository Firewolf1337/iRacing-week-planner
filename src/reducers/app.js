import moment, { duration } from 'moment';
import { CHANGE_MODAL, UPDATE_DAYS } from '../actions/app';
import { SIGNED_IN } from '../actions/auth';
import { seasonStart, weekSeasonStart, seasonEnd } from '../config';

const currentDate = moment(new Date()).utc().startOf('day');
const currentDays = currentDate.diff(seasonStart, 'days');
const currentDate2 = moment(new Date()).utc().startOf('day');
const currentDays2 = currentDate2.diff(seasonStart, 'days');
const getWeek = (date) => Math.ceil(duration(moment(date).add({ second: 1 }).diff(weekSeasonStart)).asWeeks());


export default function app(
  state = {
    date: currentDate,
    date2: currentDate2,
    daysSinceSeasonStart: currentDays,
    daysSinceSeasonStart2: currentDays2,
    week: getWeek(currentDate),
    week2: getWeek(currentDate2),
    currentModal: null,
  },
  { type, days, modalName },
) {
  if (type === UPDATE_DAYS) {
    const date = moment(seasonStart).add(days[0], 'days');
    const date2 = moment(seasonStart).add(days[1], 'days');
    return {
      ...state,
      date: date,
      date2: date2,
      daysSinceSeasonStart: days[0],
      daysSinceSeasonStart2: days[1],
      week: getWeek(date),
      week2: getWeek(date2),
    };
  }

  if (type === CHANGE_MODAL) {
    return {
      ...state,
      currentModal: modalName,
    };
  }

  if (type === SIGNED_IN) {
    return {
      ...state,
      currentModal: null,
    };
  }

  return state;
}
