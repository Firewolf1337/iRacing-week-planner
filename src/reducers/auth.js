import { initializeApp } from 'firebase/app';

import {
  LOADING_AUTH, SIGNED_OUT, SIGNED_IN, LOADING_RESET, RESET_SENT, ERROR_AUTH, ERROR_ACKNOWLEDGE, ERROR_RESET,
} from '../actions/auth';
import { firebaseConfig } from '../config';

const firebaseApp = initializeApp(firebaseConfig);

export default function auth(
  state = {
    user: null,
    firebaseApp,
    loadingAuth: true,
    loadingReset: false,
    errorAuth: null,
    errorReset: null,
  },
  { type, user, error },
) {
  if (type === SIGNED_OUT) {
    return { ...state, user: null };
  }

  if (type === LOADING_AUTH) {
    return { ...state, loadingAuth: true };
  }

  if (type === ERROR_AUTH) {
    return { ...state, loadingAuth: false, errorAuth: error };
  }

  if (type === ERROR_RESET) {
    return { ...state, loadingReset: false, errorReset: error };
  }

  if (type === ERROR_ACKNOWLEDGE) {
    return { ...state, errorAuth: null, errorReset: null };
  }

  if (type === SIGNED_IN) {
    return { ...state, loadingAuth: false, errorAuth: null, user };
  }

  if (type === LOADING_RESET) {
    return { ...state, loadingReset: true };
  }

  if (type === RESET_SENT) {
    return { ...state, loadingReset: false, errorReset: null };
  }

  return state;
}
