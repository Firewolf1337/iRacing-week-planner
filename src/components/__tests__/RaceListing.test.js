import { describe, test, beforeEach, afterEach } from '@jest/globals';
import moment from 'moment';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MockDate from 'mockdate';
import { UPDATE_SETTING } from '../../actions/settings';

import RaceListing from '../RaceListing';
import { defaultFilters } from '../../reducers/settings';

import '../../data/season.json';
import '../../data/racelengths.json';
import '../../data/racetimes.json';

jest.mock('../../data/season.json');
jest.mock('../../data/racelengths.json');
jest.mock('../../data/racetimes.json');

const mockStore = configureMockStore([thunk]);

const createNodeMock = (element) => {
  if (element.type === 'td') {
    return { clientHeight: 50 };
  }

  return null;
};

describe('components/RaceListing', () => {
  const defaultStore = {
    settings: {
      sort: { key: 'id', order: 'asc' },
      filters: defaultFilters,
      favouriteSeries: [],
      ownedTracks: [],
      ownedCars: [],
      favouriteCars: [],
      favouriteTracks: [],
      columns: ['id'],
    },
    app: {
      date: moment('2020-09-23T01:00:00.000Z'),
    },
  };

  beforeEach(() => {
    MockDate.set('2020-09-23T13:30:00.000Z');
  });

  afterEach(() => {
    MockDate.reset();
  });

  const getMockStore = (settingsOverrides = {}, filtersOverrides = {}) => mockStore({
    ...defaultStore,
    settings: {
      ...defaultStore.settings,
      ...settingsOverrides,
      filters: {
        ...defaultStore.settings.filters,
        ...filtersOverrides,
      },
    },
  });

  const getSortExpectation = (sortVars) => expect.objectContaining({
    type: UPDATE_SETTING,
    payload: expect.objectContaining({
      key: 'sort',
      value: expect.objectContaining(sortVars),
    }),
  });

  test('renders correctly', () => {
    const component = renderer.create(<Provider store={getMockStore()}><RaceListing /></Provider>);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('changes sort on click', () => {
    const store = getMockStore({
      sort: { key: 'id', order: 'asc' },
      columns: ['id', 'series'],
    });

    const component = renderer.create(<Provider store={store}><RaceListing /></Provider>, { createNodeMock });

    expect(component.toJSON()).toMatchSnapshot();

    component.root.findByProps({ id: 'raceListing-th-id' }).children[0].props.onClick();

    expect(store.getActions()[0]).toEqual(getSortExpectation({ key: 'id', order: 'desc' }));

    component.root.findByProps({ id: 'raceListing-th-series' }).children[0].props.onClick();

    expect(store.getActions()[1]).toEqual(getSortExpectation({ key: 'series', order: 'asc' }));
  });

  test('changes descending sort on click', () => {
    const store = getMockStore({
      sort: { key: 'series', order: 'desc' },
      columns: ['id', 'series', 'raceTimes'],
    });
    const component = renderer.create(<Provider store={store}><RaceListing /></Provider>, { createNodeMock });
    expect(component.toJSON()).toMatchSnapshot();

    component.root.findByProps({ id: 'raceListing-th-series' }).children[0].props.onClick();

    expect(store.getActions()[0]).toEqual(getSortExpectation({ key: 'series', order: 'asc' }));

    expect(component.root.findByProps({ id: 'raceListing-th-raceTimes' }).children[0]).toBeString();
  });

  test('renders not found favourite series', () => {
    const store = getMockStore({ favouriteSeries: [543] }, { favouriteSeries: true });

    const component = renderer.create(<Provider store={store}><RaceListing /></Provider>);

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.root.findAllByType('table')).toEqual([]);
  });

  test('renders not found favourite cars', () => {
    const store = getMockStore({ favouriteCars: [9999] }, { favouriteCarsOnly: true });

    const component = renderer.create(<Provider store={store}><RaceListing /></Provider>);

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.root.findAllByType('table')).toEqual([]);
  });

  test('renders not found favourite tracks', () => {
    const store = getMockStore({ favouriteTracks: [9999] }, { favouriteTracksOnly: true });

    const component = renderer.create(<Provider store={store}><RaceListing /></Provider>);

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.root.findAllByType('table')).toEqual([]);
  });
});
