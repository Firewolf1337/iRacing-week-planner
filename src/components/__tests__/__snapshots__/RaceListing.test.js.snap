// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RaceListing changes descending sort on click 1`] = `
<div
  className="table-responsive raceListing"
>
  <table
    className="table"
    style={
      Object {
        "fontSize": "0.8em",
      }
    }
  >
    <thead>
      <tr>
        <th
          className="clickableCell"
          id="raceListing-th-id"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            ID
            <span>
               
            </span>
          </button>
        </th>
        <th
          className="clickableCell"
          id="raceListing-th-series"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            Series
            <span>
               
            </span>
            <span
              className="glyphicon glyphicon-triangle-bottom"
            />
          </button>
        </th>
        <th
          className={null}
          id="raceListing-th-raceTimes"
        >
          Race times
          <span>
             
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div>
            167
          </div>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            <span>
               
            </span>
            iRacing ARCA Menards Series
          </button>
        </td>
        <td>
          <div>
            Every 30 minutes starting at 0:00 UTC
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            32
          </div>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            <span>
               
            </span>
            iRacing Advanced Legends Cup
          </button>
        </td>
        <td>
          <div>
            Every 2 hours starting at 1:55 UTC
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            432
          </div>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            <span>
               
            </span>
            Fanatec GT Challenge
          </button>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            Set Times
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            139
          </div>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            <span>
               
            </span>
            Fanatec Global Mazda MX-5 Cup
          </button>
        </td>
        <td>
          <div>
            Every hour starting at 0:30 UTC
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/RaceListing changes sort on click 1`] = `
<div
  className="table-responsive raceListing"
>
  <table
    className="table"
    style={
      Object {
        "fontSize": "0.8em",
      }
    }
  >
    <thead>
      <tr>
        <th
          className="clickableCell"
          id="raceListing-th-id"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            ID
            <span>
               
            </span>
            <span
              className="glyphicon glyphicon-triangle-top"
            />
          </button>
        </th>
        <th
          className="clickableCell"
          id="raceListing-th-series"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            Series
            <span>
               
            </span>
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div>
            32
          </div>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            <span>
               
            </span>
            iRacing Advanced Legends Cup
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            139
          </div>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            <span>
               
            </span>
            Fanatec Global Mazda MX-5 Cup
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            167
          </div>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            <span>
               
            </span>
            iRacing ARCA Menards Series
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            432
          </div>
        </td>
        <td
          className="clickableCell"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "minHeight": 50,
              }
            }
            type="button"
          >
            <span>
               
            </span>
            Fanatec GT Challenge
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/RaceListing renders correctly 1`] = `
<div
  className="table-responsive raceListing"
>
  <table
    className="table"
    style={
      Object {
        "fontSize": "0.8em",
      }
    }
  >
    <thead>
      <tr>
        <th
          className="clickableCell"
          id="raceListing-th-id"
        >
          <button
            className="cellButton"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            ID
            <span>
               
            </span>
            <span
              className="glyphicon glyphicon-triangle-top"
            />
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div>
            32
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            139
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            167
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            432
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/RaceListing renders not found favourite cars 1`] = `
<p>
  No races this week match your favourite cars. Try turning the filter off or adding some.
</p>
`;

exports[`components/RaceListing renders not found favourite series 1`] = `
<p>
  No races this week match your favourite series. Try turning the filter off or adding some.
</p>
`;

exports[`components/RaceListing renders not found favourite tracks 1`] = `
<p>
  No races this week match your favourite tracks. Try turning the filter off or adding some.
</p>
`;
